ctp2_code/ai/diplomacy/nproposalevent.cpp

```diff
@@ -713,7 +713,7 @@ STDEHANDLER(TradePact_NewProposalEvent)
 	if (!args->GetInt(0, desireType))
 		return GEV_HD_Continue;
 
-	if ((MOTIVATION_TYPE) desireType != MOTIVATION_DESIRE_MAKE_FRIEND ||
+	if ((MOTIVATION_TYPE) desireType != MOTIVATION_DESIRE_MAKE_FRIEND &&
```

Comment:
Although I agree with you that this code make no sense. Did you test the change or reasoned that it is an improvement?

Reply:
This is a reasoned improvement. Unfortunately I am only just beginning to look at the code, so am not yet familiar enough to be certain that this is what was intended (and should happen).
I am certain the previous behaviour of always executing code in the if statement was unintentional, though.